### PR TITLE
[MIRROR] Reworks janitor cyborg cleaning, focus on the slipping [MDB IGNORE]

### PR DESCRIPTION
**Pull request by SkyratBot**
Original PR: https://github.com/tgstation/tgstation/pull/64131
--------------------

## About The Pull Request

Alt of #64105 and #64126 (I'm sorry Novva, I should have said something earlier)
I main janitor. As a janitor main, my greatest joy in life is slipping people who ignore my signs

I've seen some people complain about janitor borgs, so I decided to look into em

Unfortunately, not only is the janitor borg just a straight upgrade to janitors, it has absolutely no reason to use most of its kit
We give it standard cleaning supplies, and hell even bespoke tools to deal with leaving slippery tiles everywhere, but we also just let them clean anything they can walk over

This seems a bit too much to me, even for borgs. Also it's like, really boring

So what if we made their movement based cleaning cost something? How about we make it suck water from their bucket. Use the same pattern as mop code, make it twice as expensive though. Give it a slowdown, some sound cues, and an action button to trigger it all

## Why It's Good For The Game

We're left with a little engine of hate that can rev up to deal with large messes, or use its soap or mop to deal with smaller stuff, or to conserve water. There's tradeoffs instead of one size fits all solution, and something more unique then just a roundstart pimping ride upgrade (It's also shit, don't worry I'm coming for you too janitor mains)

(Slowdown is something close to 2.5 tiles a second, which feels roughly right to me. Video may seem slower, I'm coding on a cpu from 2007 with no graphics card, you'll have to forgive me)
 
https://user-images.githubusercontent.com/58055496/149652724-c16a63d6-8f9d-415a-bda5-2eb704157107.mp4

Oh and of course they can in theory splash anything, since the concept of a jannieborg puttering along making a hallway lubbed is very funny to me

I like this a lot better then lmao jannieborg go brr. Lemme know what you think yeah?

Oh and since they're going to need to actually use water now, I've traded out the borg's dense holosign projector for just a standard nondense one. Deploys instantly instead of having a 2 second do_after.

Ah and I threw some minor reagent code cleanup in too, to make my job a bit easier

(I can't tell if the blue stuff at the borg's feet is water or flaps, and I think they look significantly worse without it so I will choose to believe in flaps)
## Changelog

:cl:
add: Added an action button to janitor cyborgs. It'll draw water from your bucket and clean the ground under you as you walk. More wasteful then mopping mind
del: Removed the janitor cyborg's autoclean feature
balance: Swapped out the janitor cyborg's dense holosign projector for a non dense version. It doesn't have the 2 second wait attached to it
refactor: Made being a cleaning chemical into a bitflag instead of having manual checks in mop code
/:cl:

### CODE REVIEW

**Comment by Zergspower** [comment]
this mans never used a spray bottle in his lifetime

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -156,8 +156,12 @@
 	glass_name = "glass of water"
 	glass_desc = "The father of all refreshments."
 	shot_glass_icon_state = "shotglassclear"
+<<<<<<< HEAD
 	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED
 	evaporates = TRUE //SKYRAT EDIT ADDITION
+=======
+	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED|REAGENT_CLEANS
+>>>>>>> 8f32cbe38d9 (Reworks janitor cyborg cleaning, focus on the slipping (#64131))
```
Comment (GoldenAlpharex) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	chemical_flags = REAGENT_CAN_BE_SYNTHESIZED|REAGENT_CLEANS
	evaporates = TRUE //SKYRAT EDIT ADDITION
```

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

**Review by GoldenAlpharex — APPROVED** [review]
Yes

**Comment by chizzyyyy** [comment]
good pr, the windup is fucking funny lmao

**Pull request merged**